### PR TITLE
Better mass cancels with LUTs

### PR DESCRIPTION
**Pull request by filipzeta**
LUTs let us pack way more ixs into a single tx. This PR increases the amount of cancels we can do at the same time.

### CODE REVIEW

**Comment by kelvinlau20100** [comment]
we can also add up to 13 place orders in a single tx. with LUTs on a single market

File: src/client.ts [hunk 1]
```diff
@@ -638,14 +664,37 @@ export class Client {
     if (asset != undefined) {
       return await this.getSubClient(asset).cancelAllOrdersNoError();
     } else {
-      let allTxIds = [];
+      // Grab all cancel instructions across all assets
+      let ixs = [];
       await Promise.all(
```
Comment (kelvinlau20100) [inline src/client.ts:669]: same here no need for async

File: src/client.ts [hunk 2]
```diff
@@ -638,14 +664,37 @@ export class Client {
...
-        this.getAllSubClients().map(async (subClient) => {
-          let txIds = await subClient.cancelAllOrdersNoError();
-          allTxIds = allTxIds.concat(txIds);
+        this.getAllSubClients().map(async (subclient) => {
+          ixs.push(subclient.cancelAllOrdersNoErrorIxs());
```
Comment (kelvinlau20100) [inline src/client.ts:671]: same issue here

File: src/client.ts [hunk 3]
```diff
@@ -624,11 +624,37 @@ export class Client {
...
-      let txIds = [];
-      for (var subClient of this.getAllSubClients()) {
-        txIds.push(await subClient.cancelAllOrders());
-      }
-      return txIds.flat();
+      // Grab all cancel instructions across all assets
+      let ixs = [];
+      await Promise.all(
+        this.getAllSubClients().map(async (subclient) => {
+          ixs.push(subclient.cancelAllOrdersIxs());
```
[the remark this answers is not in the document]
Comment (filipzeta) [reply src/client.ts]: fixed - shouldve been concat

File: src/client.ts [hunk 4]
```diff
@@ -624,11 +624,37 @@ export class Client {
...
       return await this.getSubClient(asset).cancelAllOrders();
     } else {
-      let txIds = [];
-      for (var subClient of this.getAllSubClients()) {
-        txIds.push(await subClient.cancelAllOrders());
-      }
-      return txIds.flat();
+      // Grab all cancel instructions across all assets
+      let ixs = [];
+      await Promise.all(
```
[the remark this answers is not in the document]
Comment (filipzeta) [reply src/client.ts]: good catch - removed in latest commit

**Comment by filipzeta** [comment]
Verified tx limits on mainnet with a script that sends many orders and then attempts to cancel them all: 
image

Upper limit for `cancelAllOrders()` is 13 across all assets and 18 if only one asset is used. We'll set 13 in constants so it's conservative (maybe a bit lower to account for potential future assets?)

`cancelAllPerpMarketOrders()` works fine for all 5 assets, and the TX size is 1492 without LUTs so should have plenty of overhead for extra assets.

**Review by kelvinlau20100 — APPROVED** [review]
LGTM

**Pull request merged**